src/display/screen.js

```diff
@@ -49,11 +49,11 @@ Screen.prototype.fillImageData = function(buffer) {
             for (var x = 0; x < Screen.physics.WIDTH; x++) {
                 for (var px = 0; px < this.pixelSize; px++) {
                     var offset = yOffset + (x * this.pixelSize + px);
-                    var v = Screen.colors[buffer[y * Screen.physics.WIDTH + x]];
+                    var v = Screen.colors[buffer[y * Screen.physics.WIDTH + x]] || [];
```

Comment:
an empty array would fail anyway on the next line, so I wouldn't bother with this fallback
(a validation of the Screen.colors format during init could be a better option)

Reply:
I added this fallback because I had the following error else:

```
Uncaught TypeError: Cannot read property '0' of undefined
```

Reply:
ok, the issue is that "buffer[y * Screen.physics.WIDTH + x]" is undefined when the GPU is bootstraped until the game actually writes to the screen

can you update with the following line instead ?
`var v = Screen.colors[buffer[y * Screen.physics.WIDTH + x] | 0];`

Reply:
Sure, done. :)